ls1intum/Artemis
# `Development`: Add client tests for timeframe view

### Checklist

#### General

- [x] I tested the PR on a testserver
- [x] I chose a title conforming to
  the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client

- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following
  the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).

### Description

Follow-up for https://github.com/ls1intum/Artemis/pull/7363 adding tests and improving the coverage for the timeframe
view.

### Steps for Testing

1. Verify that tests are passing and tests make sense _(code review)_
2. Make sure the renaming of files did not break displaying the exercises
3. Test if filters and sorting does still work as expected 
4. Test if exercises are displayed properly in the `weekly view`
5. Verify that upcoming exercises are displayed properly

### Review Progress
#### Manual Tests

- [x] Manual Test 1
- [x] Manual Test 2

#### Code Review


### Test Coverage

#### Client

| Class/File | Line Coverage |   Confirmation (assert/expect) |
|------------|--------------:|-------------------------------:|
| [exercise.utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5602/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exercise/exercise.utils.ts.html) | 90.58% |                              ✅ |
| [course-exercises-grouped-by-timeframe.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5602-2/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-exercises/course-exercises-grouped-by-timeframe.component.ts.html) | 98.52%	 |                              ✅ |
| [course-exercises-grouped-by-week.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5602/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-exercises/course-exercises-grouped-by-week.component.ts.html) | 97.77% |                              ✅ |
| [course-exercises.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5602/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-exercises/course-exercises.component.ts.html) | 98.02% |                              ✅ |
